cw-orch-daemon/src/state.rs

```diff
@@ -148,7 +183,14 @@ impl DaemonState {
     }
     /// Get the state filepath and read it as json
     fn read_state(&self) -> Result<serde_json::Value, DaemonError> {
-        crate::json_file::read(&self.json_file_path)
+        // Check if already open in write mode {
+        let lock = GLOBAL_WRITE_STATE.lock().unwrap();
```

Comment:
so this freezes the thread if the file is open in read mode?

Reply:
> so this freezes the thread if the file is open in read mode?

Yes, it still should be faster if other thread have it open in write mode, and slightly slower otherwise

Reply:
added drop of guard in case of reading a file